lib/Backend/Lower.cpp

```diff
@@ -594,7 +594,17 @@ Lowerer::LowerRange(IR::Instr *instrStart, IR::Instr *instrEnd, bool defaultDoFa
 
         case Js::OpCode::CloneStr:
         {
-            GenerateGetImmutableOrScriptUnreferencedString(instr->GetSrc1()->AsRegOpnd(), instr, IR::HelperOp_CompoundStringCloneForAppending, false);
+            IR::RegOpnd * strOpnd = instr->GetSrc1()->AsRegOpnd();
+
+            // Check if strOpnd is NULL before the CloneStr. There could be cases where SimpleJit might have dead stored instructions corresponding
+            // to the definition of strOpnd and as a result interpreter stack has strOpnd as nullptr. During FullJit we may not dead store the instructions
+            // defining strOpnd due to StSlot instructions added at the end of jitted loop body. As a result, when we bailout (BailOnSimpleJitToFullJitLoopBody)
+            // strOpnd could have a NULL value causing CloneStr to dereference a nullptr.
+            auto strOpndIsNull = IR::LabelInstr::New(Js::OpCode::Label, this->m_func);
```

Comment:
I expected your change to be in the runtime side instead of the JIT'd code.  If we leave it in the JIT'd code, we should only do it for JIT loop bodies, but I kind of think it's just better to put this check in the helper itself.  It minimizes the amount of code the JIT has to generate, which is better throughput wise and memory wise.

Reply:
As discussed offline, we do have a vtable check to avoid the helper call when not necessary that needs this to be in JIT'd code. I will add a check to do this only for JIT loop bodies. Btw, for some reason ChakraHub isn't helping me update this PR so I will abandon this and create a new one.